xcp_d/config.py

```diff
@@ -793,3 +798,88 @@ def to_filename(filename):
     """Write settings to file."""
     filename = Path(filename)
     filename.write_text(dumps())
+
+
+def dismiss_hash(entities: list | None = None):
+    """Set entities to dismiss in a DerivativesDataSink."""
+    entities = entities or []
+    output_layout = execution.output_layout
+    if output_layout != 'multiverse':
+        entities.append('hash')
+    return entities
+
+
+DEFAULT_DISMISS_ENTITIES = dismiss_hash()
+
+DEFAULT_CONFIG_HASH_FIELDS = {
+    'execution': [
+        'bids_filters',
+        'confounds_config',
+        'atlases',
+        'output_layout',
+    ],
+    'workflow': [
+        'mode',
+        'file_format',
+        'dummy_scans',
+        'input_type',
+        'despike',
+        'smoothing',
+        'output_interpolated',
+        'combine_runs',
+        'motion_filter_type',
+        'band_stop_min',
+        'band_stop_max',
+        'motion_filter_order',
+        'head_radius',
+        'fd_thresh',
+        'min_time',
+        'bandpass_filter',
+        'high_pass',
+        'low_pass',
+        'bpf_order',
+        'min_coverage',
+        'correlation_lengths',
+        'process_surfaces',
+        'abcc_qc',
+        'linc_qc',
+    ],
+}
+
+
+def hash_config(
+    conf: dict[str, ty.Any],
+    *,
+    fields_required: dict[str, list[str]] = DEFAULT_CONFIG_HASH_FIELDS,
+    version: str = None,
+    digest_size: int = 4,
+) -> str:
+    """
+    Generate a unique BLAKE2b hash of configuration attributes.
+    By default, uses a preselected list of workflow-altering parameters.
+    """
+    import json
+    from hashlib import blake2b
+    from pprint import pprint
+
+    if version is None:
+        from xcp_d import __version__ as version
+
+    # Load the preprocessing derivatives dataset description
+    pprint(conf)
+    dset_desc_path = Path(conf['execution']['fmri_dir']) / 'dataset_description.json'
```

Comment:
I suppose technically we should use the hashes from all of the linked datasets, but I don't think the HBCD/ABCD folks care about hashes outside of the preprocessing pipeline.